### ISSUE TITLE
Support for D3js 4.x

### ISSUE THREAD

**Issue by ganeshchand** [spaced/scala-js-d3#25]
Has this project been updated to support latest 4.x d3js release?

**Comment by spaced** [comment]
nope. because d3 v4 is modularized, it becomes a new repo (for each module)

**Comment by fdietze** [comment]
I'm using d3 v4 already and started to write some simple facades.
Any ideas how I can separate the the facades into different libraries where each library is exposing its methods in a global d3 object?

Seems impossible right now:
https://github.com/scala-js/scala-js/issues/1892

**Comment by fdietze** [comment]
For everyone who is interested, my current work is being done here:
https://github.com/woost/wust2/tree/master/scalajs-d3v4/src/main/scala

I will move this to a separate project in the future. Contributions and feedback welcome.

**Comment by fdietze** [comment]
I put the facades in its own project now and would like to have some feedback on the build and code architecture: https://github.com/fdietze/scala-js-d3v4 ([File-Tree](https://github.com/fdietze/scala-js-d3v4/find/master)). Especially: [Modules.scala](https://github.com/fdietze/scala-js-d3v4/blob/master/project/Modules.scala). There are many open questions, like versioning of the modules and inter-dependency handling.

**Comment by fdietze** [comment]
Update:
My D3v4 Facade is usable and waiting for contributions:
https://github.com/fdietze/scala-js-d3v4
